pkg/controller/node/node_controller.go

```diff
@@ -424,7 +432,8 @@ func (ctrl *Controller) syncMachineConfigPool(key string) error {
 	// TODO: Deep-copy only when needed.
 	pool := machineconfigpool.DeepCopy()
 	everything := metav1.LabelSelector{}
-	if reflect.DeepEqual(pool.Spec.MachineSelector, &everything) {
+
+	if reflect.DeepEqual(pool.Spec.MachineSelector, &everything) && reflect.DeepEqual(pool.Spec.NodeSelector, &everything) {
```

Comment:
we should still do the "if MachineSelector is empty, get and use NodeSelector" and only compare a selector which can be either (like you did in the other places)

Reply:
for this one i was a little confused, bc it's just checking to see if MachineSelector is empty, but we shouldnt' proceed into this if if Machine Selector is empty and NodeSelector is not.

Reply:
the nil check and use NodeSelector instead is below in starting at line 452

Reply:
but we can still do

selector
if machineselector empty
   selector = nodeSelector
if reflect.DeepEqual(selector) ...

for new mcp, machineSelector is gonna be empty and nodeSelector is still correctly checked. For old mcp, machineselector is empty and nodeSelector is going to be empty anyway and correctly checked right?

Reply:
ok, your code is basically equivalent and better/cleaner/shorter :)